vllm/engine/llm_engine.py

```diff
@@ -609,15 +636,29 @@ def _get_stats(self,
             time_to_first_tokens = time_last_iters if prompt_run else []
             time_per_output_tokens = [] if prompt_run else time_last_iters
 
+            # Finished Requests
```

Comment:
Its a bit hard to follow where we have 5-6 loops that loop through the same list of `scheduled_seq_groups`